easybuild/easyconfigs/s/scenicplus/scenicplus-1.0a1-18072024-foss-2023a.eb

```diff
@@ -0,0 +1,281 @@
+easyblock = 'PythonBundle'
+
+name = 'scenicplus'
+version = '1.0a1-18072024'
+local_commit = "fa55dae"
+
+homepage = 'https://github.com/aertslab/scenicplus'
+description = """SCENIC+ is a python package to build gene regulatory networks (GRNs)
+using combined or separate single-cell gene expression (scRNA-seq) and single-cell
+chromatin accessibility (scATAC-seq) data."""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+builddependencies = [
+    ('poetry', '1.5.1'),
+    ('maturin', '1.4.0', '-Rust-1.75.0'),
+    ('CMake', '3.26.3'),
+]
+dependencies = [
+    ('Python', '3.11.3'),
+    ('Python-bundle-PyPI', '2023.06'),
+    ('SciPy-bundle', '2023.07'),
+    ('Single-cell-python-bundle', '2024.02'),
+    ('anndata', '0.10.5.post1'),
+    ('BeautifulSoup', '4.12.2'),
+    ('h5py', '3.9.0'),
+    ('python-igraph', '0.11.4'),
+    ('imageio', '2.33.1'),
+    ('leidenalg', '0.10.2'),
+    ('lxml', '4.9.2'),
+    ('MACS2', '2.2.9.1'),
+    ('numba', '0.58.1'),
+    ('polars', '0.20.2'),
+    ('Arrow', '14.0.1'),
+    ('plotly.py', '5.16.0'),
+    ('pybedtools', '0.9.1'),
+    ('pyBigWig', '0.3.22'),
+    ('pyGAM', '0.9.1'),
+    ('Pysam', '0.22.0'),
+    ('Ray-project', '2.9.1'),
+    ('scikit-image', '0.22.0'),
+    ('snakemake', '8.4.2'),
+    ('statsmodels', '0.14.1'),
+    ('IPython', '8.14.0'),
+    ('Pandoc', '3.1.2', '', SYSTEM),
+    ('ipympl', '0.9.3'),
+    ('PyTables', '3.8.0'),
+    ('Sphinx-RTD-Theme', '2.0.0'),
+    ('scrublet', '0.2.3'),
+    ('numexpr', '2.9.0'),
+    ('Kaleido', '0.2.1'),
+    ('pyfasta', '0.5.2'),
+    ('gensim', '4.3.2'),
+]
+
+# fix requirements.txt of scenicplus
+local_preinstallopts = (
+    "sed -i"
+    " -e '/typing==3.7.4.3/d'"
+    " -e 's/==.*//g'"
+    " -e '/pyarrow-hotfix/d'"
+    " -e '/line-profiler/d'"
+    " -e '/ndindex/d'"
+    " -e '/sinfo/d'"
+    " requirements.txt && "
+)
+# unpin version restriction and delete typing requirement
+local_pycistopic_preinstallopts = (
+    "sed -i"
+    " -e 's/pyscenic>=0.12.0/pyscenic/'"
+    " -e 's/pandas==1.5/pandas/'"
+    " -e 's/python-Levenshtein/Levenshtein/'"
+    " -e 's/python-igraph/igraph/'"
+    " -e '/typing/d'"
+    # delete pyscenic and loomxpy req from commit - have versions now
+    " -e '60,61d'"
```

Comment:
Can we avoid relying on line numbers here, and use something like `/loomxpy/d` instead?